### PR TITLE
Add: Locomotive Engineer Exercise

### PR DESCRIPTION
**Pull request by meatball133**

### CODE REVIEW

File: exercises/concept/locomotive-engineer/.meta/design.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Design
+
+## Goal
+
+This concept exercise is meant to teach an understanding/use of **decomposition** and the `*` (splat) and `**` (double splat) operators in Ruby.
+
+<br>
```
Comment (iHiD) [inline exercises/concept/locomotive-engineer/.meta/design.md:7]:
```suggestion
```

File: concepts/multiple-assignment-and-decomposition/links.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (kotp) [inline concepts/multiple-assignment-and-decomposition/links.json:1]:
```suggestion
[]

```

File: exercises/concept/locomotive-engineer/.docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+
+numbers = {a: 1, b: 2, c: 3}
+my_method(**numbers)
+3
+2
+1
+```
+
+[args]: https://docs.ruby-lang.org/en/3.1/syntax/methods_rdoc.html#label-Array-2FHash+Argument
+[kwargs]: https://docs.ruby-lang.org/en/3.1/syntax/methods_rdoc.html#label-Keyword+Arguments
```
Comment (kotp) [inline exercises/concept/locomotive-engineer/.docs/introduction.md:415]: We use full words for other reference links, not sure why we are using abbreviations for these two.

**Review by kotp — APPROVED** [review]
Looking pretty nice.  Good commit history, approving now, as it appears there are only slight changes needed before it can get tested in "real" terms.

**Pull request merged**